packages/apps/dashboard/server/src/modules/details/details.service.ts

```diff
@@ -148,40 +153,51 @@ export class DetailsService {
     return result;
   }
 
-  public async getLeadersByChainId(
-    chainId?: ChainId,
-    take?: number,
+  public async getLeaders(
+    chainId: ChainId,
+    { orderBy, orderDirection, first }: GetLeadersPaginationOptions = {},
   ): Promise<LeaderDto[]> {
-    const chainIds = !chainId
-      ? await this.networkConfig.getAvailableNetworks()
-      : [chainId];
+    const filter = this.createLeadersFilter(
+      chainId,
+      orderBy,
+      orderDirection,
+      first,
+    );
 
-    let allLeadersData: ILeader[] = [];
-    for (const id of chainIds) {
-      const leadersData = await OperatorUtils.getLeaders({ chainId: id });
-      allLeadersData = allLeadersData.concat(
-        leadersData.filter((leader) => leader.amountStaked > 0 && leader.role),
-      );
-    }
+    const [rawLeaders, reputations] = await Promise.all([
```

Comment:
I think this is not going to work. Lets say we want to get 4 leaders ordered by reputation, then:
- OperatorUtils.getLeaders(filter) returns leaders A, B, C, D.
- fetchReputations return B, C, E, F.
Then we would be only returning B and C, when we should be getting B,C,E,F